RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesAdapter.java

```diff
@@ -28,24 +34,44 @@ public ItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
 
     @Override
     public void onBindViewHolder(ItemViewHolder holder, final int position) {
-        String stateText;
-        stateText = states.get(position);
+        final StateItem item = states.get(position);
+        String stateText = item.title;
 
         final Context context = holder.itemView.getContext();
         holder.textView.setText(stateText);
 
+        TypedArray images = context
+                .getResources()
+                .obtainTypedArray(R.array.states_images);
```

Comment:
If you write a log you can see in `logcat monitor` in Android studio. eg `Log.d(TAG, "onBindViewHolder");`.
We can see that `onBindViewHolder ` is called often.

In this case getting an array everytime might not be best for performance.
It might be a good idea to pass the array in the constructor. ie
`public StatesAdapter(ArrayList<StateItem> states, TypedArray images)`